lib/elixir/lib/code/formatter.ex

```diff
@@ -1438,15 +1440,17 @@ defmodule Code.Formatter do
     {bitstring_wrap_parens(doc, i, last), state}
   end
 
-  defp bitstring_spec_to_algebra({op, _, [left, right]}, state, normalize_modifiers)
+  defp bitstring_spec_to_algebra({op, _, [left, right]}, state, normalize_modifiers, paren_op)
        when op in [:-, :*] do
     normalize_modifiers = normalize_modifiers && op != :*
-    {left, state} = bitstring_spec_to_algebra(left, state, normalize_modifiers)
+    {left, state} = bitstring_spec_to_algebra(left, state, normalize_modifiers, op)
     {right, state} = bitstring_spec_element_to_algebra(right, state, normalize_modifiers)
-    {concat(concat(left, Atom.to_string(op)), right), state}
+    doc = concat(concat(left, Atom.to_string(op)), right)
+    doc = if paren_op == :* && op != :*, do: wrap_in_parens(doc), else: doc
```

Comment:
I didn't add the test because I didn't know if this was a concern or not, but this would change the following (probably irrelevant) behavior:

Current (same in main):

```elixir
iex> Code.format_string!("<<0::1*2*5>>") |> IO.puts
<<0::1*2*5>>
```

With suggestion:

```elixir
iex> Code.format_string!("<<0::1*2*5>>") |> IO.puts
<<0::(1*2)*5>>
```

If this change is OK, will commit suggestion 🙂

Reply:
Yeah, it is ok, I would prefer so there is no ambiguity about the sides!

Reply:
To be clear, let’s do the proposed change to always add parens and then ship it!

Reply:
Thanks for confirming!
Will add a test too so it's clear it is intentional during future changes.